### PR TITLE
Update to Swift 4.2

### PR DESCRIPTION
**Pull request by zhenwei12138**
update to Swift 4.2 
change scrollView disable user interaction when show skeleton
befor:
func disableUserInteraction() { isScrollEnabled = false }
    func enableUserInteraction() { isScrollEnabled = true }
now:
func disableUserInteraction() { isUserInteractionEnabled = false }
    func enableUserInteraction() { isUserInteractionEnabled = true }

if tableView show skeleton user click cell  tableViewDelegate method Will called

### CODE REVIEW

File: Sources/Collections/CollectionSkeletonProtocol.swift [hunk 1]
```diff
@@ -28,6 +28,6 @@ extension CollectionSkeleton where Self: UIScrollView {
     var estimatedNumberOfRows: Int { return 0 }
     func addDummyDataSource() {}
     func removeDummyDataSource(reloadAfter: Bool) {}
-    func disableUserInteraction() { isScrollEnabled = false }
-    func enableUserInteraction() { isScrollEnabled = true }
+    func disableUserInteraction() { isUserInteractionEnabled = false }
+    func enableUserInteraction() { isUserInteractionEnabled = true }
```
Comment (BarredEwe) [inline Sources/Collections/CollectionSkeletonProtocol.swift]: The keyboard will appear.

Comment (zhenwei12138) [reply Sources/Collections/CollectionSkeletonProtocol.swift]: if  isScrollEnabled = true   cell will response touch  it will response tableView delegate

**Comment by felixsolorzano** [comment]
This breaks compatibility with swift 4.0, why don't you support various swift versions instead?

**Comment by Juanpe** [comment]
> This breaks compatibility with swift 4.0, why don't you support various swift versions instead?

You can install the pod version which you want, so, if you are using swift 4, you can install SkeletonView 1.4, `pod 'Objection', '0.9'`

**Comment by Juanpe** [comment]
Hi @zhenwei12138, thanks for the PR, but SkeletonView is already compatible with swift 4.2 :)

**Pull request closed without merging**